lib/Driver/OutputFileMap.cpp

```diff
@@ -119,6 +124,26 @@ bool OutputFileMap::parse(std::unique_ptr<llvm::MemoryBuffer> Buffer) {
   if (!Map)
     return true;
 
+  auto resolvePath =
+      [workingDirectory](
+          llvm::yaml::ScalarNode *Path,
+          llvm::SmallVectorImpl<char> &PathStorage) -> StringRef {
+    StringRef PathStr = Path->getValue(PathStorage);
+    if (workingDirectory.empty() || PathStr.empty() || PathStr == "-" ||
+        llvm::sys::path::is_absolute(PathStr)) {
+      return PathStr;
+    }
+    // Copy the path to avoid making assumptions about how getValue deals with
+    // Storage.
+    SmallString<128> PathStrCopy(PathStr);
+    PathStorage.clear();
+    PathStorage.reserve(PathStrCopy.size() + workingDirectory.size() + 1);
+    PathStorage.insert(PathStorage.begin(), workingDirectory.begin(),
+                       workingDirectory.end());
+    llvm::sys::path::append(PathStorage, PathStrCopy);
+    return StringRef(PathStorage.data(), PathStorage.size());
```

Comment:
Unless I'm misunderstanding, I think you're returning a StringRef to a local buffer that goes out of scope here (i.e. invalid memory).

Reply:
Whoops, yes, I misread.